### PR TITLE
Fix middleware bug

### PR DESCRIPTION
**Pull request by DevHaoZi**
Closes goravel/goravel#123

### CODE REVIEW

File: route/gin_group.go [hunk 1]
```diff
@@ -111,7 +111,6 @@ func (r *GinGroup) getGinRoutesWithMiddlewares() gin.IRoutes {
 	middlewares = append(middlewares, ginOriginMiddlewares...)
 	middlewares = append(middlewares, ginMiddlewares...)
 	middlewares = append(middlewares, ginLastMiddlewares...)
-	r.middlewares = []httpcontract.Middleware{}
```
Comment (hwbrzzl) [inline route/gin_group.go:114]: So sorry, just saw this PR! If we remove this line directly, maybe we will get the error below:
```
facades.Route.Middleware(A).Get()
facades.Route.Middleware(B).Post() // will apply middleware A and B at the same time
```

We can add a test case to confirm this.

Comment (DevHaoZi) [reply route/gin_group.go:114]: > So sorry, just saw this PR! If we remove this line directly, maybe we will get the error below:
> 
> ```
> facades.Route.Middleware(A).Get()
> facades.Route.Middleware(B).Post() // will apply middleware A and B at the same time
> ```
> 
> We can add a test case to confirm this.

Any suggestions for improvement? I'm not very familiar with Gin's middleware usage.

Comment (hwbrzzl) [reply route/gin_group.go:114]: Maybe we can move this line to a new method: `func (r *GinGroup) clearMiddlewares()`, then call the method at the end of methods of calling `r.getGinRoutesWithMiddlewares()`: `Get`, `Post`, etc.
```
func (r *GinGroup) Any(relativePath string, handler httpcontract.HandlerFunc) {
	r.getGinRoutesWithMiddlewares().Any(pathToGinPath(relativePath), []gin.HandlerFunc{handlerToGinHandler(handler)}...)
	r.clearMiddlewares()
}

func (r *GinGroup) Resource(relativePath string, controller httpcontract.ResourceController) {
	r.getGinRoutesWithMiddlewares().GET(pathToGinPath(relativePath), []gin.HandlerFunc{handlerToGinHandler(controller.Index)}...)
	r.getGinRoutesWithMiddlewares().POST(pathToGinPath(relativePath), []gin.HandlerFunc{handlerToGinHandler(controller.Store)}...)
	r.getGinRoutesWithMiddlewares().GET(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Show)}...)
	r.getGinRoutesWithMiddlewares().PUT(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Update)}...)
	r.getGinRoutesWithMiddlewares().PATCH(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Update)}...)
	r.getGinRoutesWithMiddlewares().DELETE(pathToGinPath(relativePath+"/{id}"), []gin.HandlerFunc{handlerToGinHandler(controller.Destroy)}...)
	r.clearMiddlewares()
}
```

Comment (DevHaoZi) [reply route/gin_group.go:114]: ok, fixed it.

File: route/gin_group.go [hunk 2]
```diff
@@ -5,20 +5,20 @@ import (
 
 	"github.com/gin-gonic/gin"
 
-	httpcontract "github.com/goravel/framework/contracts/http"
+	contractshttp "github.com/goravel/framework/contracts/http"
```
Comment (hwbrzzl) [inline route/gin_group.go:8]: S: I saw it's httpcontract in Laravel, so I think we don't need to change it, and we can change to this in other places in the future.
image

Comment (hwbrzzl) [reply route/gin_group.go:8]: Emm, I think we should have a unified standard on it, so `contractshttp ` is ok, let's do it in other places!

File: route/gin_group.go (hunk 1 above)
Comment (hwbrzzl) [reply route/gin_group.go:114]: Great! Could you please add a test case to cover this situation?
```
facades.Route.Middleware(A).Get()
facades.Route.Middleware(B).Post() // will apply middleware A and B at the same time
```

**Review by hwbrzzl — APPROVED** [review]
Great!

**Review by hwbrzzl — APPROVED** [review]
Great!

**Pull request merged**